acceptance/tests/pxp-module-puppet/run_puppet_twice.rb

```diff
@@ -0,0 +1,95 @@
+require 'pxp-agent/test_helper.rb'
+require 'puppet/acceptance/environment_utils'
+
+test_name 'Run Puppet while a Puppet Agent run is in-progress, wait for completion' do
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  env_name = test_file_name = File.basename(__FILE__, '.*')
+  environment_name = mk_tmp_environment(env_name)
+
+  step 'On master, create a new environment that will result in a slow run' do
+    site_manifest = "#{environmentpath}/#{environment_name}/manifests/site.pp"
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  case $::osfamily {
+    'windows': { exec { 'sleep':
+                        command => 'true',
+                        unless  => 'sleep 10', #PUP-5806
+                        path    => 'C:\\cygwin64\\bin',} }
```

Comment:
This looks like it will fail on 32bit Windows (and so will fail in the puppet-agent pipelines).
Note that the existing acceptance test that has this exec resource does also have this problem already, but that test is currently skipped on all Windows variants.